### PR TITLE
LFN 0x7160/0 truename fix and updated tests

### PR DESCRIPTION
**Pull request by andrewbird**
Essentially we were upcasing the whole of the returned string, when we should have left the last component alone. Checked against Win98 for correctness.

### CODE REVIEW

File: src/dosext/mfs/lfn.c [hunk 1]
```diff
@@ -1076,17 +1076,22 @@ static int mfs_lfn_(void)
...
 			build_ufs_path(fpath, filename, drive);
 			if (!find_file(fpath, &st, get_redirection_root1(drive, NULL, 0), &doserrno))
 				return lfn_error(doserrno);
 			make_unmake_dos_mangled_path(filename, fpath, drive, 2 - _CL);
 		} else {
-			strupperDOS(filename);
+			char *lastslash = strrchr(filename, '\\');
+			if (lastslash) { /* Upcase but preserve the final component */
+				*lastslash = '\0';
+				strupperDOS(filename);
```
Comment (stsp) [inline src/dosext/mfs/lfn.c:1088]: Maybe strnupperDOS() instead?
But oh well.

**Comment by stsp** [comment]
Thanks!

File: src/dosext/mfs/lfn.c (hunk 1 above)
Comment (andrewbird) [reply src/dosext/mfs/lfn.c:1088]: > strnupperDOS()

Did you just invent that?

Comment (stsp) [reply src/dosext/mfs/lfn.c:1088]: No, that was a proposal for this patch.

Comment (andrewbird) [reply src/dosext/mfs/lfn.c:1088]: Ahh, I see.

**Pull request merged**